src/main/java/io/cdap/plugin/http/source/common/pagination/page/JsonPage.java

```diff
@@ -43,17 +40,19 @@ class JsonPage extends BasePage {
   private final BaseHttpSourceConfig config;
   private final List<String> optionalFields;
 
+  // TODO : handle case where the result json object is null
```

Comment:
what happens when result json object is null? are you planning to handle that case in another PR?